mlserver/handlers/dataplane.py

```diff
@@ -90,6 +97,7 @@ async def infer(
         ).count_exceptions()
 
         with infer_duration, infer_errors:
+            cache_key = payload.json()
```

Comment:
Depending on the request size, this could be quite expensive - would it make sense to only run it when caching enabled?

Alternatively, would it make sense to use the request ID if present?

Reply:
So the idea here is that if the cache key is independent of a request ID and other headers, we can expect more cache hits. No two requests should have the same ID so then we don't benefit from caching. I understand this is expensive, and a different choice of cache key would probably make sense. Let me know your thoughts.

Reply:
Ah, I see what you mean - that makes complete sense.

In that case, it may be worth only computing the `payload.json()` key when the response cache is enabled.

Reply:
Do you think it is a good idea to compress the cache key string?

Reply:
Good idea - although I would keep it simple for now, and just use the JSON payloads you have now. We can always come back to it if we see memory spikes.